pages/activityList.js

```diff
@@ -79,22 +80,36 @@ const ErrorMessage = styled.p`
 
 export default function ActivityList({ activities }) {
   const [searchTerm, setSearchTerm] = useState("");
+  const [selectedByIcon, setSelectedByIcon] = useState([]);
+  console.log(selectedByIcon);
 
-  const getFilteredActivities = (searchTerm, activities) => {
-    if (!searchTerm) {
-      return activities;
-    }
-
-    return activities.filter(
+  const getFilteredActivities = () => {
+    let filtered = activities.filter(
       (activity) =>
         activity.category.toLowerCase().includes(searchTerm.toLowerCase()) ||
         activity.title.toLowerCase().includes(searchTerm.toLowerCase()) ||
         activity.area.toLowerCase().includes(searchTerm.toLowerCase()) ||
         activity.country.toLowerCase().includes(searchTerm.toLowerCase())
     );
+
+    if (selectedByIcon.length > 0) {
+      filtered = filtered.filter((activity) =>
+        selectedByIcon.includes(activity.categoryFilter)
+      );
+    }
+
+    return filtered;
   };
 
-  const filteredActivities = getFilteredActivities(searchTerm, activities);
+  const filteredActivities = getFilteredActivities();
```

Comment:
if you don't plan to use the function somewhere else I would suggest to remove the function and put the code directly into your component. This is a bit less convoluted.

Reply:
This is a point that is not implementable as we had a problem yesterday where the activities list was shown twice up to the point when we implemented the `const getFilteredActivities()`in this way. That is why this function needs to stay here. 😊